From fa699e2c4b08fed331606328f1b6dbd3410318ed Mon Sep 17 00:00:00 2001
From: Gaganpreet kaur <gaganpreetx.kaur@intel.com>
Date: Mon, 2 Jul 2018 14:51:34 +0530
Subject: [PATCH 6/6] Fix for Bluetooth device name is resetting to default
 name after reboot

Reason: When the BT device name is updated is getting saved
to the config data pointer but not to the persistent data
(i.e. bt_config.conf).So, when the reboot is happening it
is not able to get the updated device name from the persistent
data (i.e. bt_config.conf) as during reboot bt_config_save is
not called. It is only called on the BT Enable Event.

Fix: Saving the BT Device name to persistent data using
btif_config_save once it is set.

Change-Id: I1e001114d00cffc78db0c10d859d39d799e1d425
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-65735
Signed-off-by: anitha3x <anithax.h.chandrasekar@intel.com>
---
 btif/src/btif_storage.cc | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/btif/src/btif_storage.cc b/btif/src/btif_storage.cc
index 9d4a84e..0a9754a 100644
--- a/btif/src/btif_storage.cc
+++ b/btif/src/btif_storage.cc
@@ -203,8 +203,10 @@ static int prop2cfg(const RawAddress* remote_bd_addr, bt_property_t* prop) {
       value[prop->len] = '\0';
       if (remote_bd_addr)
         btif_config_set_str(bdstr, BTIF_STORAGE_PATH_REMOTE_NAME, value);
-      else
+      else {
         btif_config_set_str("Adapter", BTIF_STORAGE_KEY_ADAPTER_NAME, value);
+        btif_config_save();
+      }
       break;
     case BT_PROPERTY_REMOTE_FRIENDLY_NAME:
       strncpy(value, (char*)prop->val, prop->len);
-- 
1.9.1

